<launch>
  <arg name="ip_address" default="192.168.11.1" />
  <arg name="raw_ladar_data" default="false" />

  <node pkg="slamware_ros_sdk" exec="slamware_ros_sdk_server_node" name="slamware_ros_sdk_server_node" output="both">

  <param name="ip_address"               value="$(var ip_address)"/>
  <param name="angle_compensate"         value="true"/>

  <param name="map_frame"                value="slamware_map"/>
  <param name="robot_frame"              value="base_link"/>
  <param name="odom_frame"               value="odom"/>
  <param name="laser_frame"              value="laser"/>
  <param name="imu_frame"                value ="imu_link"/>
  <param name="camera_left"              value ="camera_left"/>
  <param name="camera_right"             value ="camera_right"/>

  <param name="robot_pose_pub_period"    value="0.05"/>
  <param name="scan_pub_period"          value="0.1"/>
  <param name="map_pub_period"           value="0.2"/>
  <param name="imu_raw_data_period"      value="0.005"/>

  <param name= "ladar_data_clockwise"    value= "true"/>
  <param name = "robot_pose_topic"            value = "robot_pose"/>
  
  <!-- topic remap /-->
  <remap from="scan"                       to="scan"/>
  <remap from="odom"                       to="odom"/>
  <remap from="map"                        to="slamware_map"/>
  <remap from="map_metadata"               to="map_metadata"/>
  </node>

  <node pkg="tf2_ros" exec="static_transform_publisher" name="odom2map" args="0 0 0 0 0 0 1 slamware_map odom"/>
  <node pkg="tf2_ros" exec="static_transform_publisher" name="laser2base" args="0 0 0.0315 0 0 0 1 base_link laser"/>
  <node pkg="tf2_ros" exec="static_transform_publisher" name="leftcam2base" args="0.0418 0.03 0 -0.5 0.5 -0.5 0.5 base_link camera_left"/>
  <node pkg="tf2_ros" exec="static_transform_publisher" name="rightcam2Leftcam" args="0.06 0 0 0 0 0 1 camera_left camera_right"/>
  <node pkg="tf2_ros" exec="static_transform_publisher" name="imu2Leftcam" args="0.03 0 0 0 0 -0.7071068 0.7071068 camera_left imu_link"/>
</launch>
